Crashlytics/ProtoSupport/Protos/crashlytics.proto

```diff
@@ -0,0 +1,252 @@
+
+    // Event type, one of "crash", "ndk-crash", "error"
+    required string type = 2;
+
+    // Application state captured at event time
+    message Application {
+      message Execution {
+        // Thread state
+        message Thread {
+          // Thread name from OS
+          optional string name = 1;
+
+          // Bitfield of FrameImportance
+          required uint32 importance = 2;
+
+          // Contents of each stack frame
+          message Frame {
+            required uint64 pc = 1;
+            optional string symbol = 2;                // On macOS
+            optional uint64 offset = 4;                // On macOS
+
+            // Bitfield of FrameImportance
+            optional uint32 importance = 5;
+          }
+          repeated Frame frames = 3;
+
+          message Register {
+            required string name = 1;
+            required uint64 value = 2;
+          }
+          repeated Register registers = 4;
+
+          optional string alternate_name = 5;          // On macOS
+        }  // message Application.Execution.Thread
+
+        repeated Thread threads = 1;
+
+        // Signal caught by SDK signal handler
+        message Signal {
+          required string name = 1;
+
+          required string code = 2;
+
+          // Faulting instruction or address
+          required uint64 address = 3;
+        }
+        required Signal signal = 3;
+
+        // Binary images/libraries included in the application package
+        message BinaryImage {
+          required uint64 base_address = 1;
+          required uint64 size = 2;
+
+          // Full path name to binary image
+          required string name = 3;
+
+          // 128-bit object UUID matches Mach-O DWARF dSYM files */
+          optional bytes uuid = 4;
+
+          optional Architecture arch = 5;
+        }
+        repeated BinaryImage binaries = 4;
+      }  // message Application.Execution
+
+      repeated CustomAttribute custom_attributes = 2;
+
+      optional bool background = 3; /* true if the app was in the background */
+      optional uint32 ui_orientation = 4;
+    }  // message Event.Application
+
+    optional Application app = 3;
+
+    // Device physical properties recorded at event time
+    message Device {
+      // Battery level from 0.0 to 1.0
+      optional float battery_level = 1;
+
+      // True when the proximity sensor is activated during the event
+      optional bool proximity_on = 3;
+
+      // Device orientation during event
+      // 1,2 = portrait
+      // 3,4 = landscape
+      // 5 = face up
+      // 6 = face down
+      optional uint32 orientation = 4;
+
+      // Total RAM used at event time, in bytes
+      optional uint64 ram_used = 5;
+
+      // Total disk used at event time, in bytes
+      optional uint64 disk_used = 6;
+    }
+
+    optional Device device = 5;
+
+    // Developer-supplied log lines
+    // Only those lines most recently logged prior to the event are captured
+    message Log {
```

Comment:
Is this being used?

Reply:
Yes, this is needed for our Crashlytics.log() API.